2025-03-09 22:19:47,951 INFO    Thread-11 (_run_job):196 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221947-lsmu08i7\logs\debug.log
2025-03-09 22:19:47,953 INFO    Thread-11 (_run_job):196 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221947-lsmu08i7\logs\debug-internal.log
2025-03-09 22:19:47,954 INFO    Thread-11 (_run_job):196 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:19:47,954 INFO    Thread-11 (_run_job):196 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 32, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 4, 'optimizer': 'rmsprop', 'weightDecay': 0, 'weightInit': 'random'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:19:47,955 INFO    Thread-11 (_run_job):196 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:19:48,079 INFO    Thread-11 (_run_job):196 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/lsmu08i7
2025-03-09 22:19:48,080 ERROR   Thread-11 (_run_job):196 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:19:48,080 INFO    Thread-11 (_run_job):196 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:19:48,080 INFO    Thread-11 (_run_job):196 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:19:48,080 INFO    Thread-11 (_run_job):196 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:19:48,080 INFO    Thread-11 (_run_job):196 [wandb_run.py:_restore():2322] restore
2025-03-09 22:19:48,080 INFO    Thread-11 (_run_job):196 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:19:50,127 INFO    Thread-11 (_run_job):196 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:19:50,127 INFO    Thread-11 (_run_job):196 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:19:50,127 INFO    Thread-11 (_run_job):196 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:19:50,134 ERROR   MainThread:196 [pyagent.py:_run_jobs_from_queue():234] Run lsmu08i7 errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\4244206215.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\4244206215.py", line 133, in backpropagation
    deltas.append((deltas[-1].dot(self.weights[i].T)).T * self.sigmoid_derivative(self.layers[i]))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (32,116) and (32,32) not aligned: 116 (dim 1) != 32 (dim 0)

2025-03-09 22:19:55,154 INFO    MsgRouterThr:196 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_setup.py:_flush():67] Configure stats pid to 196
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_setup.py:_flush():67] Loading settings from C:\Users\prajv\.config\wandb\settings
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_setup.py:_flush():67] Loading settings from c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\settings
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221956-5hsij4z4\logs\debug.log
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221956-5hsij4z4\logs\debug-internal.log
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:monkeypatch_ipython():599] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x000001F8330B2C00>
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 132, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 5, 'optimizer': 'nadam', 'weightDecay': 0.5, 'weightInit': 'xavier'}
config: {'_wandb': {}}
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():784] starting backend
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():788] sending inform_init request
2025-03-09 22:19:56,184 INFO    Thread-16 (_run_job):196 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=spawn, using: spawn
2025-03-09 22:19:56,192 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():798] backend started and connected
2025-03-09 22:19:56,193 INFO    Thread-16 (_run_job):196 [wandb_run.py:_config_callback():1261] config_cb None None {'activationFunc': 'tanh', 'batchSize': 132, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 5, 'optimizer': 'nadam', 'weightDecay': 0.5, 'weightInit': 'xavier'}
2025-03-09 22:19:56,194 INFO    Thread-16 (_run_job):196 [wandb_run.py:_label_probe_notebook():1204] probe notebook
2025-03-09 22:19:56,216 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():891] updated telemetry
2025-03-09 22:19:56,261 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-09 22:19:58,029 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():990] starting run threads in backend
2025-03-09 22:19:58,318 INFO    Thread-16 (_run_job):196 [wandb_run.py:_console_start():2375] atexit reg
2025-03-09 22:19:58,318 INFO    Thread-16 (_run_job):196 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-09 22:19:58,318 INFO    Thread-16 (_run_job):196 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-09 22:19:58,319 INFO    Thread-16 (_run_job):196 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-09 22:19:58,320 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-09 22:19:58,329 INFO    Thread-16 (_run_job):196 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221958-5hsij4z4\logs\debug.log
2025-03-09 22:19:58,329 INFO    Thread-16 (_run_job):196 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221958-5hsij4z4\logs\debug-internal.log
2025-03-09 22:19:58,330 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:19:58,330 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 132, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 5, 'optimizer': 'nadam', 'weightDecay': 0.5, 'weightInit': 'xavier'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:19:58,330 INFO    Thread-16 (_run_job):196 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:19:58,457 INFO    Thread-16 (_run_job):196 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/5hsij4z4
2025-03-09 22:19:58,458 ERROR   Thread-16 (_run_job):196 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:19:58,458 INFO    Thread-16 (_run_job):196 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:19:58,458 INFO    Thread-16 (_run_job):196 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:19:58,458 INFO    Thread-16 (_run_job):196 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:19:58,458 INFO    Thread-16 (_run_job):196 [wandb_run.py:_restore():2322] restore
2025-03-09 22:19:58,458 INFO    Thread-16 (_run_job):196 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:20:00,273 INFO    Thread-16 (_run_job):196 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:20:00,273 INFO    Thread-16 (_run_job):196 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:20:00,274 INFO    Thread-16 (_run_job):196 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:20:00,285 ERROR   MainThread:196 [pyagent.py:_run_jobs_from_queue():234] Run 5hsij4z4 errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\4244206215.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\4244206215.py", line 133, in backpropagation
    deltas.append((deltas[-1].dot(self.weights[i].T)).T * self.sigmoid_derivative(self.layers[i]))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (64,132) and (64,64) not aligned: 132 (dim 1) != 64 (dim 0)

2025-03-09 22:20:10,343 INFO    MsgRouterThr:196 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_setup.py:_flush():67] Configure stats pid to 196
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_setup.py:_flush():67] Loading settings from C:\Users\prajv\.config\wandb\settings
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_setup.py:_flush():67] Loading settings from c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\settings
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_222011-s1chmnn2\logs\debug.log
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_222011-s1chmnn2\logs\debug-internal.log
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:monkeypatch_ipython():599] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x000001F833094650>
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 32, 'learningRate': 0.0001, 'num_epochs': 10, 'num_hiddenLayers': 3, 'optimizer': 'adam', 'weightDecay': 0.0005, 'weightInit': 'random'}
config: {'_wandb': {}}
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():784] starting backend
2025-03-09 22:20:11,243 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():788] sending inform_init request
2025-03-09 22:20:11,252 INFO    Thread-21 (_run_job):196 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=spawn, using: spawn
2025-03-09 22:20:11,252 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():798] backend started and connected
2025-03-09 22:20:11,254 INFO    Thread-21 (_run_job):196 [wandb_run.py:_config_callback():1261] config_cb None None {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 32, 'learningRate': 0.0001, 'num_epochs': 10, 'num_hiddenLayers': 3, 'optimizer': 'adam', 'weightDecay': 0.0005, 'weightInit': 'random'}
2025-03-09 22:20:11,255 INFO    Thread-21 (_run_job):196 [wandb_run.py:_label_probe_notebook():1204] probe notebook
2025-03-09 22:20:11,274 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():891] updated telemetry
2025-03-09 22:20:11,336 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-09 22:20:13,002 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():990] starting run threads in backend
2025-03-09 22:20:13,280 INFO    Thread-21 (_run_job):196 [wandb_run.py:_console_start():2375] atexit reg
2025-03-09 22:20:13,281 INFO    Thread-21 (_run_job):196 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-09 22:20:13,281 INFO    Thread-21 (_run_job):196 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-09 22:20:13,281 INFO    Thread-21 (_run_job):196 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-09 22:20:13,283 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-09 22:20:13,286 INFO    Thread-21 (_run_job):196 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_222013-s1chmnn2\logs\debug.log
2025-03-09 22:20:13,286 INFO    Thread-21 (_run_job):196 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_222013-s1chmnn2\logs\debug-internal.log
2025-03-09 22:20:13,286 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:20:13,286 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 32, 'learningRate': 0.0001, 'num_epochs': 10, 'num_hiddenLayers': 3, 'optimizer': 'adam', 'weightDecay': 0.0005, 'weightInit': 'random'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:20:13,286 INFO    Thread-21 (_run_job):196 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:20:13,471 INFO    Thread-21 (_run_job):196 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/s1chmnn2
2025-03-09 22:20:13,473 ERROR   Thread-21 (_run_job):196 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:20:13,473 INFO    Thread-21 (_run_job):196 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:20:13,473 INFO    Thread-21 (_run_job):196 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:20:13,473 INFO    Thread-21 (_run_job):196 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:20:13,473 INFO    Thread-21 (_run_job):196 [wandb_run.py:_restore():2322] restore
2025-03-09 22:20:13,473 INFO    Thread-21 (_run_job):196 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:20:15,541 INFO    Thread-21 (_run_job):196 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:20:15,541 INFO    Thread-21 (_run_job):196 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:20:15,541 INFO    Thread-21 (_run_job):196 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:20:15,549 ERROR   MainThread:196 [pyagent.py:_run_jobs_from_queue():234] Run s1chmnn2 errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\4244206215.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_196\4244206215.py", line 133, in backpropagation
    deltas.append((deltas[-1].dot(self.weights[i].T)).T * self.sigmoid_derivative(self.layers[i]))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (32,116) and (32,32) not aligned: 116 (dim 1) != 32 (dim 0)

2025-03-09 22:20:15,553 ERROR   MainThread:196 [pyagent.py:_run_jobs_from_queue():245] Detected 3 failed runs in the first 60 seconds, killing sweep.
